iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/TerminalRefXml.java

```diff
@@ -18,21 +20,52 @@
  */
 public final class TerminalRefXml {
 
+    /**
+     * @deprecated Use {@link #writeTerminalRef(Connectable, Terminal, NetworkXmlWriterContext, String)} instead
+     */
+    @Deprecated
     public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String elementName) throws XMLStreamException {
-        writeTerminalRef(t, context, context.getVersion().getNamespaceURI(), elementName);
+        writeTerminalRef(null, t, context, context.getVersion().getNamespaceURI(), elementName);
     }
 
+    public static void writeTerminalRef(Connectable owner, Terminal t, NetworkXmlWriterContext context, String elementName) throws XMLStreamException {
+        writeTerminalRef(owner, t, context, context.getVersion().getNamespaceURI(), elementName);
+    }
+
+    /**
+     * @deprecated Use {@link #writeTerminalRef(Connectable, Terminal, NetworkXmlWriterContext, String, String)} instead
+     */
+    @Deprecated
     public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName) throws XMLStreamException {
-        writeTerminalRef(t, context, namespace, elementName, context.getWriter());
+        writeTerminalRef(null, t, context, namespace, elementName, context.getWriter());
+    }
+
+    public static void writeTerminalRef(Connectable owner, Terminal t, NetworkXmlWriterContext context, String namespace, String elementName) throws XMLStreamException {
+        writeTerminalRef(owner, t, context, namespace, elementName, context.getWriter());
     }
 
+    /**
+     * @deprecated Use {@link #writeTerminalRef(Connectable, Terminal, NetworkXmlWriterContext, String, String, XMLStreamWriter)} instead
+     */
+    @Deprecated
     public static void writeTerminalRef(Terminal t, NetworkXmlWriterContext context, String namespace, String elementName, XMLStreamWriter writer) throws XMLStreamException {
+        writeTerminalRef(null, t, context, namespace, elementName, writer);
+    }
+
+    public static void writeTerminalRef(Connectable owner, Terminal t, NetworkXmlWriterContext context, String namespace, String elementName, XMLStreamWriter writer) throws XMLStreamException {
         Connectable c = t.getConnectable();
         if (!context.getFilter().test(c)) {
             throw new PowsyblException("Oups, terminal ref point to a filtered equipment " + c.getId());
         }
         writer.writeEmptyElement(namespace, elementName);
-        writer.writeAttribute("id", context.getAnonymizer().anonymizeString(c.getId()));
+        IidmXmlUtil.runUntilMaximumVersion(IidmXmlVersion.V_1_2, context, () ->
+                XmlUtil.writeStringAttribute("id", context.getAnonymizer().anonymizeString(c.getId()), writer)
+        );
+        IidmXmlUtil.runFromMinimumVersion(IidmXmlVersion.V_1_3, context, () -> {
+            if (!c.equals(owner)) {
```

Comment:
Two connectable could be equals, but different: for example if there are in two different networks

Reply:
In this case, does it really matter ?
If they are equals, ids are same even if they are in two differents networks.

Reply:
Yes, this is a bit reaching but I think you should do `c != owner` to prevent mistakes in any case.